test/AutoDiff/validation-test/method.swift

```diff
@@ -0,0 +1,395 @@
+  }
+}
+
+extension Parameter : Differentiable, AdditiveArithmetic {
+  typealias TangentVector = Parameter
+  typealias Scalar = Tracked<Float>
+  typealias Shape = ()
+  init(repeating repeatedValue: Tracked<Float>, shape: ()) {
+    self.init(x: repeatedValue)
+  }
+  static func + (lhs: Parameter, rhs: Parameter) -> Parameter {
+    return Parameter(x: lhs.x + rhs.x)
+  }
+  static func - (lhs: Parameter, rhs: Parameter) -> Parameter {
+    return Parameter(x: lhs.x - rhs.x)
+  }
+  static func * (lhs: Scalar, rhs: Parameter) -> Parameter {
+    return Parameter(x: lhs * rhs.x)
+  }
+  static var zero: Parameter { return Parameter(x: 0) }
+}
+
+MethodTests.testWithLeakChecking(
+  "instance method with generated adjoint, called from differentated func"
+) {
+  func f(_ p: Parameter) -> Tracked<Float> {
+    return 100 * p.squared()
+  }
+  expectEqual(Parameter(x: 4 * 100), gradient(at: Parameter(x: 2), in: f))
+  expectEqual(Parameter(x: 40 * 100), gradient(at: Parameter(x: 20), in: f))
+}
+
+MethodTests.testWithLeakChecking(
+  "instance method with generated adjoint, differentiated directly"
+) {
+  // This is our current syntax for taking gradients of instance methods
+  // directly. If/when we develop nicer syntax for this, change this test.
+  func g(p: Parameter) -> Tracked<Float> { p.squared() }
+  expectEqual(Parameter(x: 4), gradient(at: Parameter(x: 2), in: g))
+  expectEqual(Parameter(x: 40), gradient(at: Parameter(x: 20), in: g))
+}
+
+MethodTests.testWithLeakChecking("instance method with generated adjoint, wrt only self") {
+  func f(_ p: Parameter) -> Tracked<Float> {
+    return 100 * p.multiplied(with: 200)
+  }
+  expectEqual(Parameter(x: 100 * 200), gradient(at: Parameter(x: 1), in: f))
+  expectEqual(Parameter(x: 100 * 200), gradient(at: Parameter(x: 2), in: f))
+}
+
+MethodTests.testWithLeakChecking("instance method with generated adjoint, wrt only non-self") {
+  func f(_ other: Tracked<Float>) -> Tracked<Float> {
+    return 100 * Parameter(x: 200).multiplied(with: other)
+  }
+  expectEqual(100 * 200, gradient(at: 1, in: f))
+  expectEqual(100 * 200, gradient(at: 2, in: f))
+}
+
+// FIXME: Add a binary differential operator.
+//
+// MethodTests.testWithLeakChecking(
+//   "instance method with generated adjoint, wrt self and non-self"
+// ) {
+//   let g = #gradient({ (p: Parameter, o: Tracked<Float>) in p.multiplied(with: o) })
+//   expectEqual((Parameter(x: 100), 200), g(Parameter(x: 200), 100))
+//   expectEqual((Parameter(x: 200), 100), g(Parameter(x: 100), 200))
+// }
+
+MethodTests.testWithLeakChecking(
+  "static method with generated adjoint, called from differentiated func"
+) {
+  func f(_ p: Parameter) -> Tracked<Float> {
+    return 100 * Parameter.squared(p: p)
+  }
+  expectEqual(Parameter(x: 4 * 100), gradient(at: Parameter(x: 2), in: f))
+  expectEqual(Parameter(x: 40 * 100), gradient(at: Parameter(x: 20), in: f))
+}
+
+// TODO(SR-8699): Fix this test.
+// MethodTests.testWithLeakChecking(
```

Comment:
I believe `StdlibUnitTest` has a nice built-in way of XFAILing tests. Does this not consistently fail/pass/crash?